src/libsystemd-network/network-internal.c

```diff
@@ -101,11 +102,46 @@ static bool net_condition_test_strv(char * const *patterns, const char *string)
         return has_positive_rule ? match : true;
 }
 
+static int net_condition_test_property(char * const *match_property, sd_device *device) {
+        char * const *p;
+
+        if (strv_isempty(match_property))
+                return true;
+
+        STRV_FOREACH(p, match_property) {
+                _cleanup_free_ char *key = NULL;
+                const char *val, *dev_val;
+                bool invert, v;
+
+                invert = **p == '!';
```

Comment:
So "!" is present in every element in the strv? Yeah, I see that config_parse_match_property() does that.

Reply:
Yes.